src/Optimizely/DecisionService/DecisionService.php

```diff
@@ -148,6 +150,152 @@ public function getVariation(Experiment $experiment, $userId, $attributes = null
   }
 
   /**
+   * Get the variation the user is bucketed into for the given FeatureFlag
+   * @param  FeatureFlag $featureFlag The feature flag the user wants to access
+   * @param  string      $userId      user id
+   * @param  array       $attributes  user attributes
+   * @return array/null  {"experiment" : Experiment, "variation": Variation } / null
+   */
+  public function getVariationForFeature(FeatureFlag $featureFlag, $userId, $attributes){
+    //Evaluate in this order: 
+    //1. Attempt to bucket user into all experiments in the feature flag.
+    //2. Attempt to bucket user into rollout in the feature flag.
+
+    // Check if the feature flag is under an experiment and the the user is bucketed into one of these experiments
+    $result = $this->getVariationForFeatureExperiment($featureFlag, $userId, $attributes);
+    if($result)
+      return $result;
+
+    // Check if the feature flag has rollout and the user is bucketed into one of it's rules
+    $variation = $this->getVariationForFeatureRollout($featureFlag, $userId, $attributes);
+    if($variation){
+      $this->_logger->log(Logger::INFO,
+        "User '{$userId}' is in the rollout for feature flag '{$featureFlag->getKey()}'."
+      );
+      
+      return array(
+        "experiment" => null,
+        "variation" => $variation);
+      
+    } else{
+      $this->_logger->log(Logger::INFO,
+        "User '{$userId}' is not in the rollout for feature flag '{$featureFlag->getKey()}'."
```

Comment:
plz. follow java code. msgs looks from ruby.

Reply:
I have replaced messages from Java where appropriate. Edited few messages as well because of an issue in Java msgs. Shouldn't create an issue